pkg/virt-controller/watch/export/export.go

```diff
@@ -0,0 +1,866 @@
+	} else {
+		return service.(*corev1.Service), nil
+	}
+}
+
+func (ctrl *VMExportController) createServiceManifest(vmExport *exportv1.VirtualMachineExport) *corev1.Service {
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ctrl.getExportServiceName(vmExport),
+			Namespace: vmExport.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(vmExport, schema.GroupVersionKind{
+					Group:   exportv1.SchemeGroupVersion.Group,
+					Version: exportv1.SchemeGroupVersion.Version,
+					Kind:    "VirtualMachineExport",
+				}),
+			},
+			Labels: map[string]string{
+				virtv1.AppLabel: exportv1.App,
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Protocol: "TCP",
+					Port:     443,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 8443,
+					},
+				},
+			},
+			Selector: map[string]string{
+				exportServiceLabel: vmExport.Name,
+			},
+		},
+	}
+	return service
+}
+
+func (ctrl *VMExportController) getExporterPod(vmExport *exportv1.VirtualMachineExport) (*corev1.Pod, bool, error) {
+	key := controller.NamespacedKey(vmExport.Namespace, ctrl.getExportPodName(vmExport))
+	if obj, exists, err := ctrl.PodInformer.GetStore().GetByKey(key); err != nil {
+		log.Log.V(3).Errorf("error %v", err)
+		return nil, false, err
+	} else if !exists {
+		return nil, exists, nil
+	} else {
+		pod := obj.(*corev1.Pod)
+		return pod, exists, nil
+	}
+}
+
+func (ctrl *VMExportController) createExporterPod(vmExport *exportv1.VirtualMachineExport, pvcs []*corev1.PersistentVolumeClaim) (*corev1.Pod, error) {
+	log.Log.V(3).Infof("Checking if pod exist: %s/%s", vmExport.Namespace, ctrl.getExportPodName(vmExport))
+	key := controller.NamespacedKey(vmExport.Namespace, ctrl.getExportPodName(vmExport))
+	if obj, exists, err := ctrl.PodInformer.GetStore().GetByKey(key); err != nil {
+		log.Log.V(3).Errorf("error %v", err)
+		return nil, err
+	} else if !exists {
+		manifest := ctrl.createExporterPodManifest(vmExport, pvcs)
+
+		log.Log.V(3).Infof("Creating new exporter pod %s/%s", manifest.Namespace, manifest.Name)
+		ctrl.Recorder.Eventf(vmExport, corev1.EventTypeNormal, exporterPodCreatedEvent, "Creating exporter pod %s/%s", manifest.Namespace, manifest.Name)
+		return ctrl.Client.CoreV1().Pods(vmExport.Namespace).Create(context.Background(), manifest, metav1.CreateOptions{})
+	} else {
+		pod := obj.(*corev1.Pod)
+		return pod, nil
+	}
+}
+
+func (ctrl *VMExportController) createExporterPodManifest(vmExport *exportv1.VirtualMachineExport, pvcs []*corev1.PersistentVolumeClaim) *corev1.Pod {
+	podManifest := ctrl.TemplateService.RenderExporterManifest(vmExport, exportPrefix)
+	podManifest.ObjectMeta.Labels = map[string]string{exportServiceLabel: vmExport.Name}
+	podManifest.Spec.SecurityContext = &corev1.PodSecurityContext{
+		RunAsUser: &[]int64{0}[0],
```

Comment:
Not sure we have to deal with this now but we may not want to do this all the time.  For example with kubevirt content type I think we can `RunAsNonRoot` and set the fsgroup to 107 or whatever it is for qemu